### ISSUE TITLE
#1214 The bug is not fixed

### ISSUE THREAD

**Issue by WaN3DDK** [yhirose/cpp-httplib#1215]
### Code
```
httplib::Server svr;
int main()
{
	svr.Get("/login", [](const httplib::Request& req, httplib::Response& res) {
		for (auto elem : req.params) {
			cout << elem.first << " " << elem.second << endl;
		}
		});
	svr.listen("127.0.0.1", 8080);
}
```

Enter the URL in your browser.
http://127.0.0.1:8080/login#token_type=Bearer&access_token=123456789&expires_in=604800
Program error
![image](https://user-images.githubusercontent.com/31143076/158637677-1785347f-e7e8-4c38-b707-7070818f9473.png)

**Comment by yhirose** [comment]
@WaN3DDK, thanks for find the out-of-bound problem. I just fixed it. Regarding 'The URL parameter is also not fetched correctly', HTTP servers are not responsible to handle a fragment string. I tested with the Python Flask server, and I got the same result. Thanks.

**Comment by WaN3DDK** [comment]
> token_type=Bearer&access_token=123456789&expires_in=604800
req.has_param("access_token") Cannot get parameters. `req.params.size =0`

**Comment by yhirose** [comment]
Yes, this is a correct behavior as I mentioned in my previous comment. Fragment text (any characters after '#') should be ignored by HTTP server. The following stackoverflow post helps you understand it more cleaerly.
https://stackoverflow.com/questions/18752995/how-to-get-access-to-the-url-segments-after-a-in-python-flask
